anvil/src/config.rs

```diff
@@ -555,119 +555,123 @@ impl NodeConfig {
         let mut fork_timestamp = None;
 
-        let (db, fork): (Arc<RwLock<dyn Db>>, Option<ClientFork>) = if let Some(eth_rpc_url) =
-            self.eth_rpc_url.clone()
-        {
-            // TODO make provider agnostic
-            let provider = Arc::new(
-                Provider::<RetryClient<Http>>::new_client(&eth_rpc_url, 10, 1000)
-                    .expect("Failed to establish provider to fork url"),
-            );
-
-            let fork_block_number = if let Some(fork_block_number) = self.fork_block_number {
-                // auto adjust hardfork if not specified
-                if self.hardfork.is_none() {
-                    let hardfork: Hardfork = fork_block_number.into();
-                    env.cfg.spec_id = hardfork.into();
-                    self.hardfork = Some(hardfork);
-                }
-
-                fork_block_number
-            } else {
-                // pick the last block number but also ensure it's not pending anymore
-                find_latest_fork_block(&provider).await.expect("Failed to get fork block number")
-            };
-
-            let block = provider
-                .get_block(BlockNumber::Number(fork_block_number.into()))
-                .await
-                .expect("Failed to get fork block");
-
-            let block = if let Some(block) = block {
-                block
-            } else {
-                if let Ok(latest_block) = provider.get_block_number().await {
-                    panic!(
-                        "Failed to get block for block number: {}\nlatest block number: {}",
-                        fork_block_number, latest_block
-                    );
-                }
-                panic!("Failed to get block for block number: {}", fork_block_number)
-            };
-
-            env.block.number = fork_block_number.into();
-            fork_timestamp = Some(block.timestamp);
-
-            // if not set explicitly we use the base fee of the latest block
-            if self.base_fee.is_none() {
-                if let Some(base_fee) = block.base_fee_per_gas {
-                    self.base_fee = Some(base_fee);
-                    env.block.basefee = base_fee;
-                    // this is the base fee of the current block, but we need the base fee of the
-                    // next block
-                    let next_block_base_fee = fees.get_next_block_base_fee_per_gas(
-                        block.gas_used,
-                        block.gas_limit,
-                        block.base_fee_per_gas.unwrap_or_default(),
-                    );
-                    // update next base fee
-                    fees.set_base_fee(next_block_base_fee.into());
+        let (db, fork): (Arc<tokio::sync::RwLock<dyn Db>>, Option<ClientFork>) =
+            if let Some(eth_rpc_url) = self.eth_rpc_url.clone() {
+                // TODO make provider agnostic
```

Comment:
Will this be an issue if you pass in a websocket fork url?

Reply:
yes, this is still blocked by object safe connections i ethers-rs

Reply:
Any idea when websocket fork URLs will be implemented?

Reply:
it's currently being worked on